modules/fluid_properties/src/userobjects/LeadBismuthFluidProperties.C

```diff
@@ -0,0 +1,386 @@
+  dcv_de = dcv_dp * dp_de + dcv_dT * dT_de;
+}
+
+Real
+LeadBismuthFluidProperties::cv_from_p_T(Real p, Real T) const
+{
+  Real rho, drho_dT, drho_dp;
+  rho_from_p_T(p, T, rho, drho_dp, drho_dT);
+  Real alpha = -drho_dT / rho;
+  Real bulk_modulus = bulk_modulus_from_p_T(p, T);
+  Real cp = cp_from_p_T(p, T);
+  return cp / (1 + alpha * alpha * bulk_modulus * T / rho / cp);
+}
+
+void
+LeadBismuthFluidProperties::cv_from_p_T(
+    Real p, Real T, Real & cv, Real & dcv_dp, Real & dcv_dT) const
+{
+  Real cp, dcp_dp, dcp_dT;
+  cp_from_p_T(p, T, cp, dcp_dp, dcp_dT);
+  cv = cv_from_p_T(p, T);
+
+  Real rho, drho_dT, drho_dp;
+  rho_from_p_T(p, T, rho, drho_dp, drho_dT);
+  Real alpha = -drho_dT / rho;
+  Real alpha_2 = alpha * alpha, alpha_3 = alpha * alpha * alpha;
+  Real dalpha_dT = drho_dT * drho_dT / rho / rho;
+  Real bulk = bulk_modulus_from_p_T(p, T);
+  Real dbulk_dT = (-1.296e-2 + 2 * 1.32e-6 * T) * 1e9;
+  Real denominator = (1 + alpha * alpha * bulk * T / rho / cp);
+  // no pressure dependence in alpha, T, bulk modulus, cp or rho
+  dcv_dp = 0;
+  dcv_dT = dcp_dT / denominator -
+           cp / denominator / denominator *
+               (2 * alpha * dalpha_dT * bulk * T / rho / cp + alpha_2 * dbulk_dT * T / rho / cp +
+                alpha_2 * bulk / rho / cp + alpha_3 * bulk * T / rho / cp -
+                dcp_dT * alpha_2 * bulk * T / rho / cp / cp);
+}
+
+Real
+LeadBismuthFluidProperties::mu_from_v_e(Real v, Real e) const
+{
+  Real temperature = T_from_v_e(v, e);
+  return 4.94e-4 * std::exp(754.1 / temperature);
+}
+
+void
+LeadBismuthFluidProperties::mu_from_v_e(
+    Real v, Real e, Real & mu, Real & dmu_dv, Real & dmu_de) const
+{
+  Real temperature, dT_dv, dT_de;
+  T_from_v_e(v, e, temperature, dT_dv, dT_de);
+  mu = mu_from_v_e(v, e);
+  dmu_dv = dT_dv * -754.1 * 4.94e-4 * exp(754.1 / temperature) / temperature / temperature;
+  dmu_de = dT_de * -754.1 * 4.94e-4 * exp(754.1 / temperature) / temperature / temperature;
+}
+
+Real
+LeadBismuthFluidProperties::k_from_v_e(Real v, Real e) const
+{
+  Real temperature = T_from_v_e(v, e);
+  return 3.284 + 1.617e-2 * temperature - 2.305e-6 * temperature * temperature;
+}
+
+void
+LeadBismuthFluidProperties::k_from_v_e(Real v, Real e, Real & k, Real & dk_dv, Real & dk_de) const
+{
+  Real temperature, dT_dv, dT_de;
+  T_from_v_e(v, e, temperature, dT_dv, dT_de);
+  k = k_from_v_e(v, e);
+  dk_dv = 1.617e-2 * dT_dv - 2 * 2.305e-6 * dT_dv * temperature;
+  dk_de = 1.617e-2 * dT_de - 2 * 2.305e-6 * dT_de * temperature;
+}
+
+Real
+LeadBismuthFluidProperties::rho_from_p_T(Real /*pressure*/, Real temperature) const
+{
+  return 11065 - 1.293 * temperature;
+}
+
+void
+LeadBismuthFluidProperties::rho_from_p_T(
+    Real pressure, Real temperature, Real & rho, Real & drho_dp, Real & drho_dT) const
+{
+  rho = rho_from_p_T(pressure, temperature);
+  drho_dp = 0;
+  drho_dT = -1.293;
+}
+
+void
+LeadBismuthFluidProperties::rho_from_p_T(const DualReal & pressure,
```

Comment:
Use "p" and "T" for pressure and temperature everywhere in this file (do same for Lead)